### PR TITLE
Use CircleCI 2.0 for post-build

### PR DESCRIPTION
**Pull request by blowmage**
Run post-build tasks in a separate job. Use CircleCI 2.0 configuration to only run this job on the master branch.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -39,6 +38,18 @@ jobs:
...
+    - image: circleci/ruby:2.5.1-stretch
+    working_directory: ~/GoogleCloudPlatform/google-cloud-ruby
+    shell: /bin/bash --login -eo pipefail
+    steps:
+    - checkout
+    - run:
+        name: Install bundle
+        command: gem update --system && gem install bundler && bundle update
+    - run: bundle exec rake circleci:post
+
```
Comment (dazuma) [inline .circleci/config.yml]: Did you intend to omit coveralls here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -25,7 +25,6 @@ jobs:
         name: Install bundle
         command: gem update --system && gem install bundler && bundle update
     - run: bundle exec rake circleci:build
-    - run: bundle exec rake test:coveralls
```
Comment (dazuma) [inline .circleci/config.yml]: Also delete the circleci:post run under test-ruby-2.5 above?

File: .circleci/config.yml (hunk 1 above)
Comment (blowmage) [reply .circleci/config.yml]: `post` now calls `coveralls` as well as the doc build tasks.

File: .circleci/config.yml (hunk 2 above)
Comment (blowmage) [reply .circleci/config.yml]: Yeah, I thought I had done that. I will do that when I get back in front of a computer.

File: Rakefile [hunk 3]
```diff
@@ -548,14 +548,10 @@ namespace :circleci do
...
-      if ENV["CIRCLE_BRANCH"] == "master"
-        Rake::Task["bundleupdate"].invoke
-        Rake::Task["jsondoc:master"].invoke
-        Rake::Task["docs:build_master"].invoke
-      end
-    end
+    Rake::Task["bundleupdate"].invoke
+    Rake::Task["jsondoc:master"].invoke
+    Rake::Task["docs:build_master"].invoke
+    Rake::Task["test:coveralls"].invoke
```
Comment (blowmage) [inline Rakefile]: `coveralls` is now called here.

File: .circleci/config.yml (hunk 2 above)
Comment (blowmage) [reply .circleci/config.yml]: Removed.

**Comment by dazuma** [comment]
Why is travis blocking this PR? I haven't seen the travis check show up in quite a while.

**Comment by blowmage** [comment]
This branch is on the main repo. Usually I push to my forked repo. Travis runs on branches here.

**Pull request merged**